lib/semantics/check-omp-structure.cc

```diff
@@ -48,23 +48,63 @@ bool OmpStructureChecker::HasInvalidWorksharingNesting(
 
 void OmpStructureChecker::CheckAllowed(OmpClause type) {
   if (!GetContext().allowedClauses.test(type) &&
-      !GetContext().allowedOnceClauses.test(type)) {
+      !GetContext().allowedOnceClauses.test(type) &&
+      !GetContext().allowedExclusiveClauses.test(type)) {
     context_.Say(GetContext().clauseSource,
         "%s clause is not allowed on the %s directive"_err_en_US,
         EnumToString(type),
         parser::ToUpperCaseLetters(GetContext().directiveSource.ToString()));
     return;
   }
-  if (GetContext().allowedOnceClauses.test(type) && FindClause(type)) {
+  if ((GetContext().allowedOnceClauses.test(type) ||
+          GetContext().allowedExclusiveClauses.test(type)) &&
+      FindClause(type)) {
     context_.Say(GetContext().clauseSource,
         "At most one %s clause can appear on the %s directive"_err_en_US,
         EnumToString(type),
         parser::ToUpperCaseLetters(GetContext().directiveSource.ToString()));
     return;
   }
+  if (GetContext().allowedExclusiveClauses.test(type)) {
+    std::vector<OmpClause> others;
+    GetContext().allowedExclusiveClauses.IterateOverMembers([&](OmpClause o) {
+      if (FindClause(o)) others.emplace_back(o);
```

Comment:
Odd, I did have braces around here as that's how I always write and it changed it to this when I ran clang-format (version 8 I think?) using the style file in the repo root. I can correct it by hand though

Reply:
I'll give that a try, but my understanding of ```AllowShortIfStatementsOnASingleLine: true``` is that it'll apply this change for any version of clang-format, because if the resulting line would be below the character limit it will move the if statement onto one line?

Reply:
Apparently clang-format won't add or remove braces, so I must have not had braces here originally and then clang-format moved the if statement to a single line. If I add braces and then run clang-format they don't get removed.